sql/localsql/nipost/poet_registration.go

```diff
@@ -17,26 +22,46 @@ type PoETRegistration struct {
 
 func AddPoetRegistration(
 	db sql.Executor,
-	nodeID types.NodeID,
 	registration PoETRegistration,
 ) error {
 	enc := func(stmt *sql.Statement) {
-		stmt.BindBytes(1, nodeID.Bytes())
+		stmt.BindBytes(1, registration.NodeId.Bytes())
 		stmt.BindBytes(2, registration.ChallengeHash.Bytes())
 		stmt.BindText(3, registration.Address)
 		stmt.BindText(4, registration.RoundID)
 		stmt.BindInt64(5, registration.RoundEnd.Unix())
 	}
-
 	if _, err := db.Exec(`
 		insert into poet_registration (id, hash, address, round_id, round_end)
 		values (?1, ?2, ?3, ?4, ?5);`, enc, nil,
 	); err != nil {
-		return fmt.Errorf("insert poet registration for %s: %w", nodeID, err)
+		return fmt.Errorf("insert poet registration for %s: %w", registration.NodeId.Bytes(), err)
```

Comment:
This will output some rubbish (binary data as text), you probably want `.String()` instead of `.Bytes()`?

Reply:
`types.NodeID` implements the `fmt.Stringer` interface. You can directly print `registration.NodeId` and don't need to convert it to bytes or a string.